File: 0026-remove-duplicates-from-sorted-array/solution.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int numDuplicate = 0;
        int curVal = nums[0];
        int idx = 0;
        while (idx < nums.size()) {
            curVal = nums[idx];
            while (idx + 1 < nums.size() && curVal == nums[idx + 1]) {
                idx++;
                numDuplicate++;
            }
            if (idx < nums.size()) {
                nums[idx - numDuplicate] = nums[idx];
            }
            idx++;
      
        }

        return nums.size() - numDuplicate;


    }
};

File: 0026-remove-duplicates-from-sorted-array/solution.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        for i in range(len(nums) - 1):
            while i < len(nums) - 1 and nums[i] == nums[i+1] :
                del nums[i+1]
        return len(nums)
        
